Апгрейд медицинской системы

kcalbCube opened the pull request:
1. Цели ПР
1.1 Усложнить медицинскую систему в плане механики. 
1.2 Сделать психиатра врачом, а не ролеплейным персонажем

2. Перефирийные изменения.
2.1 До ПР при повреждениях мозга могла произойти боль в любой части тела с силой 10.
Теперь она может произойти в разных степенях, в зависимости от повреждений мозга.
2.2 При повреждениях мозга могут появится галлюцинации.
2.3 Банка с таблетками глюкозы содержит не 21, а 10 пилюль.
2.4 Передозировка никотином теперь больше изменяет пульс, так же немного вызывает гипоксию(BREATHLOSS)
2.5 Был случайно проведен микрорефакторинг некоторых файлов в некоторых местах(span class обернут в SPAN, отформатирован код и все в таком духе.)
2.6 Инапровалин имеет небольшое противоэпилептическое воздействие(с.м. 3.6)
2.7 Noexcutite имеет небольшое противоэпилептическое воздействие(с.м. 3.6) 
2.8 В NanoMed plus теперь есть одна банка глюкозы. 

3. Основные изменения.
3.1 Теперь активность мозга отделена от его статуса. Активность изменяется в основном из-за боли, ослепления, передозировки препаратов, переломов, потери крови. Возможно добавить больше источников.
3.2 При зашкаливающих значениях активности мозга человек впадает в эпилепсию. Это состояние достаточно быстро угнетает мозг. Так же есть промежуточные состояние эпилепсии, дающие тряску и прочее.
3.3 Дизабилити EPILEPSY снижает порог для попадания в состояние эпилепсии и изредка создает ее.
3.4 Добавлен модуль психиатрии. Психические расстройства зовут себя disease. 
3.4 Добавлен дисеас schizophrenia, дающий периодические галлюцинации.
3.5 Теперь, если повреждения мозга перейдут через 75% пациент получит случайный дисеас или эпилепсию
3.6 Добавлен псилоксон - противоэпилептический препарат. Ускоряет стабилизацию мозговой активности(в обе стороны). Изготавливается из 1 части алкизина, 1 части noexcutite, 2 частей сопорифика. Из побочных эффектов седативный эффект. При передозе вызывает сильный седативный эффект и временный паралич.
3.7 Добавлен галоперидол - препарат, снимающий часть дисеасов. Из побочных эффектов замедление. При передозировке вызывает гипоксию, с небольшим шансом передоз может вызвать депрессию(с.м. 3.9). Несовместим с алкоголем.
3.8 Добавлен кальциоксон - препарат лечащий повреждения костей.
3.9 Добавлен дисеас depression. Человеку периодически будут писаться о грусти и тоске, потерянности, он будет плакать и иногда будут появляться суицидальные мысли.
3.10 Добавлен дисеас ocd - ОКР. Человека будут преследовать навязчивые идеи о грязи и прочем. Ну и чуть-чуть галлюцинаций.

close #3708
close #3685
close #3691


- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
commit

Commit:
commit

Commit:
commit

Commit:
Merge branch 'dev' of https://github.com/kcalbCube/OnyxBay into dev

Commit:
final

Commit:
really final

Commit:
FINAL bugfix

Commit:
Balance fix

Ruforce commented:
Выглядит классно. Но зачем ты поменял ТГУИ на наноУИ? (дада я знаю, что они лучше в плане надежности работы и всё такое).

kcalbCube commented:
> Выглядит классно. Но зачем ты поменял ТГУИ на наноУИ? (дада я знаю, что они лучше в плане надежности работы и всё такое).

Я никак не разобрался как работать с тгуи. Спросил на сервере - оказывается никто не знает как. Пришлось переписать.

4310V343k commented:
> (дада я знаю, что они лучше в плане надежности работы и всё такое).

Зачем говорить если не знаешь(
ТГУИ во всём превосходит наноуи, его за тем и [портировали](https://github.com/ChaoticOnyx/OnyxBay/pull/782) даже на опух, но никто не умеет им пользоваться и его постепенно выпиливают обратно. Зачем? Потому что

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
+
+	switch(activity)
+		if(80 to 120)
+			return "good"
+		if(60 to 140)
+			return "average"
+	return "bad"
```
В одном месте сделал с большой буквы, делай везде так

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
+
+/proc/get_brain_status(var/brain_damage_procents)
+	if(brain_damage_procents >= 80)
+		return "good"
+	if(brain_damage_procents >= 50)
+		return "average"
+	return "bad"
```
В одном месте сделал с большой буквы, делай везде так

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
+
+/proc/get_blood_status(var/procents)
+	if(procents < 40)
+		return "bad"
+	if(procents < 70)
+		return "average"
+	return "good"
```
В одном месте сделал с большой буквы, делай везде так

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
+		return "average"
+	return "good"
 
 /mob/living/carbon/human/proc/get_medical_data_ui()
 	var/list/data = list()
 	var/mob/living/carbon/human/H = src
```
Че за приколы?

kcalbCube replied:
Приколы оригинального кода

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
-				data["brain_activity"] = 1 - (brain.damage / brain.max_damage)
-	else
-		data["brain_activity"] = null
+			data["brain_activity"]    = brain.get_activity()
```
Опять пробелы

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
-		data["brain_activity"] = null
+			data["brain_activity"]    = brain.get_activity()
+
+		data["brain_activity_status"] = src.get_brain_activity_status()
```
Опять src.

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
+
+		data["brain_activity_status"] = src.get_brain_activity_status()
+
+		var/brain_damage_procents = 100*(brain.damage / brain.max_damage)
+		data["brain_damage"]     	  = 100-brain_damage_procents
+		data["brain_status"]          = get_brain_status(100-brain_damage_procents)
```
Опять пробелы
```suggestion
		var/brain_damage_procents = 100 * (brain.damage / brain.max_damage)
		data["brain_damage"] = 100 - brain_damage_procents
		data["brain_status"] = get_brain_status(100 - brain_damage_procents)
```

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
 		if(H.status_flags & FAKEDEATH)
+		{
 			data["pulse"] = 0
+			data["pulse_status"] = "bad"
```
В одном месте сделал с большой буквы, делай везде так

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
 
 	if(H.should_have_organ(BP_HEART))
 		if(H.status_flags & FAKEDEATH)
+		{
```
Эти скобки тут откуда

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
-			data["pulse"] = H.get_pulse(1)
-	else
-		data["pulse"] = null
+			var/pulse     = text2num(H.get_pulse(1))
```
Проблелы

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -427,43 +413,40 @@
 		if (I.is_broken())
-			organ_data["damage"] += list("Severe")
+			organ_data += list("Moderate")
 		else if (I.is_bruised())
-			organ_data["damage"] += list("Moderate")
+			organ_data += list("Significant")
```
Мне кажется тут перепутано

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -597,4 +580,20 @@
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+	return
```
```suggestion
	return

```

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -597,4 +580,20 @@
+
+
+/obj/machinery/body_scanconsole/attack_hand(mob/user)
+	ASSERT(user)
```
Разве не моб может может тыкнуть рукой?

4310V343k commented on code/game/machinery/adv_med.dm:
```diff
@@ -597,4 +580,20 @@
+	ASSERT(connected.occupant)
+	var/data = connected.occupant.get_medical_data_ui()
+	ASSERT(data)
+	var/ui_key = "adv_med"
```
Нужно ли это?

kcalbCube replied:
Были моменты в дебаге.

4310V343k commented on code/modules/item_worth/reagents.dm:
```diff
@@ -600,4 +600,7 @@
+	value = 5
+
+/datum/reagent/psyloxon
+	value = 3.5
```
```suggestion
	value = 3.5

```

4310V343k commented on code/modules/maps/swapmaps.dm:
```diff
@@ -675,4 +680,4 @@ proc/SwapMaps_GetSize(id)
 	S["y"] >> y
 	S["z"] >> z
-	return list(x,y,z)
+	return list(x, y, z)
```
Немного не понял зачем оно в этом ПРе

4310V343k commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -539,3 +539,6 @@
 		O.rejuvenate()
+
+/mob/living/carbon/proc/add_brain_activity(activity)
+	src.internal_organs_by_name[BP_BRAIN].relative_activity += activity
```
```suggestion
	src.internal_organs_by_name[BP_BRAIN].relative_activity += activity

```

TapKo4eB replied:
```suggestion
	src.internal_organs_by_name[BP_BRAIN].relative_activity += activity

```

TobyThorne replied:
зачем 'src.'

4310V343k commented on code/modules/mob/living/carbon/human/human_helpers.dm:
```diff
@@ -196,7 +196,7 @@
 		spawn(8)
 			qdel(ping_image)
-		var/feedback = list("<span class='notice'>There are noises of movement ")
+		var/feedback = list("<span class='notice'>There are noises of movement</span>")
```
Гений. Ещё и галочку что протестил поставил. Убири

4310V343k commented on code/modules/organs/internal/brain.dm:
```diff
@@ -37,9 +45,9 @@
+	name       = "thinkpan"
+	desc       = "It looks kind of like an enormous wad of purple bubblegum."
+	icon 	   = 'icons/mob/alien.dmi'
 	icon_state = "chitin"
```
ОПЯТЬ ПРОБЕЛЫ

sweden-moose replied:
Ты не понимаешь!! Это для читабельности!!

4310V343k commented on code/modules/organs/internal/brain.dm:
```diff
@@ -48,11 +56,10 @@
-	max_damage = 100
-	if(species)
-		max_damage = species.total_health/2
+	max_damage         = species? (species.total_health/2) : 100
```
И тут пробел

4310V343k commented on code/modules/organs/internal/brain.dm:
```diff
@@ -48,11 +56,10 @@
+
 	min_bruised_damage = max_damage*0.25
-	min_broken_damage = max_damage*0.75
+	min_broken_damage  = max_damage*0.75
```
И тут тоже

4310V343k commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -115,15 +115,4 @@
-
-/datum/reagent/proc/reaction_mob(mob/target)
-	touch_mob(target)
+	. = ..()
```
```suggestion
	. = ..()

```

4310V343k commented on code/modules/reagents/reagent_containers/glass/bottle.dm:
```diff
@@ -333,3 +332,14 @@
+/obj/item/weapon/reagent_containers/glass/bottle/psyloxon/New()
+	..()
+	reagents.add_reagent(/datum/reagent/psyloxon, 60)
+	update_icon()
```
```suggestion
	update_icon()

```

TobyThorne replied:
🚫

4310V343k commented on nano/templates/adv_med.tmpl:
```diff
@@ -0,0 +1,230 @@
+ -->
+
+ <!--
+     TODO: Move this to external file
```
Да?

4310V343k reviewed (commented):
Всё

Commit:
Update code/game/machinery/adv_med.dm

Co-authored-by: 4310V343k <56449763+4310V343k@users.noreply.github.com>

kcalbCube commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,117 @@
 			hide_organs = !hide_organs
 			return TRUE
 
-/obj/machinery/body_scanconsole/ui_data(mob/user)
-	var/list/data = list()
-
-	data["connected"] = connected
-	data["medical_data"] = null
-	data["hide_status"] = hide_status
-	data["hide_organs"] = hide_organs
-
-	if (connected && connected.occupant)
-		data["medical_data"] = connected.occupant.get_medical_data_ui()
-
-	return data
-
-/obj/machinery/body_scanconsole/tg_ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
-	if(!ui)
-		ui = new(user, src, ui_key, "body_scanner", name , 500, 700, master_ui, state)
-		ui.open()
-
-/obj/machinery/body_scanconsole/attack_hand(mob/user)
-	if(..())
-		return
-	if(stat & (NOPOWER|BROKEN))
-		return
-	if(!connected || (connected.stat & (NOPOWER|BROKEN)))
-		to_chat(user, "<span class='warning'>This console is not connected to a functioning body scanner.</span>")
-		return
-	if (!(connected.occupant in connected))
-		connected.go_out()
-	if(!ishuman(connected.occupant))
-		to_chat(user, "<span class='warning'>This device can only scan compatible lifeforms.</span>")
-		return
-
-	tg_ui_interact(user)
```
Очевидно, перенесено

Commit:
fix commit

Были исправлены ошибки указанные Remind Me. Немного увеличена стандартная скорость стабилизации мозговой активности.

Commit:
Merge branch 'dev' of https://github.com/kcalbCube/OnyxBay into dev

PlinHost commented:
>ТГУИ во всём превосходит наноуи

нет

4310V343k commented:
> > ТГУИ во всём превосходит наноуи
> 
> нет

Ладно

TobyThorne commented:
![image](https://user-images.githubusercontent.com/45202681/98604895-b948c380-230e-11eb-84ce-abe03f81dd1b.png)
весь ПР в ехидных красных таблетках

Destrolaric commented:
#3691
Я предлагаю добавить автору пула решение на это предложение

sweden-moose commented on code/game/objects/items/devices/flash.dm:
```diff
@@ -85,6 +85,7 @@
 						M.drop_r_hand()
 					if(flash_strength > 5)
 						M.Weaken(2)
+					C.add_brain_activity(20)
```
Класс, только вот у нас есть еще настенные флешеры и флешбенги, с ними что?

sweden-moose replied:
Да и судя по всему, можно убить человека эпилепсией за 5 кликов, если я не прав, то поправь.

kcalbCube replied:
Исправлено

sweden-moose replied:
> Класс, только вот у нас есть еще настенные флешеры и флешбенги, с ними что?

Это точно не исправлено

kcalbCube replied:
Исправлено

sweden-moose commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -114,16 +114,4 @@
 
 /datum/reagent/Destroy() // This should only be called by the holder, so it's already handled clearing its references
-	holder = null
-	. = ..()
```
куда

4310V343k replied:
куда

TapKo4eB replied:
верни . = ..() иначе куделить нормально не будет

sweden-moose commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -635,7 +653,7 @@
 /datum/reagent/leporazine/affect_blood(mob/living/carbon/M, alien, removed)
 	if(M.bodytemperature > 310)
 		M.bodytemperature = max(310, M.bodytemperature - (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
-	else if(M.bodytemperature < 311)
+	else if(M.bodytemperature <= 310)
```
зачем?

4310V343k replied:
Надо

kcalbCube commented:
Слишком много отвратительных коммитов. Я закрою этот ПР и вскоре переоткрою: коммиты с названием "commit" и "really final" слишком сильно испортят историю коммитов. Скорее всего из-за этого ПР закроют, так что лучше сделаю это самостоятельно.

Destrolaric commented:
F

Doster-d commented:
Держу в курсе, в гите есть такая штучка как скуаш коммитов, что преобразует 1000 твоих коммитов в один красивый.

Animesku commented:
https://htmlacademy.ru/blog/boost/tools/how-to-squash-commits-and-why-it-is-needed

stylemistake commented:
ОП видимо не осилил документацию

kcalbCube commented:
Оказывается при слиянии пуш реквеста можно все коммиты объединить в один. @Doster-d @Animesku спасибо что подсказали!

Commit:
fix(disability give)

Исправил баг с тем что выдача disability была без задержки, из-за чего выдавались сразу все disability.

kcalbCube commented:
> #3691
> Я предлагаю добавить автору пула решение на это предложение
> 

Возможно добавлю кальциксон или подобное в ближайшие дни.

PlinHost committed:
brain activity from flash cap

Commit:
fix(flash only from portable device) add(calciumxon, galoperidol)

Теперь мозговая активность увеличивается с ослепления с любого флешера.
Увеличена сила антиэпилептика.
Добавлен кальциоксон - Лечит конечности.
Добавлен галоперидол - убирает дизабилити(описан как лекарство от шизофрении)

Commit:
add(Depression) fix(Calcioxon, Schizoprenia) balance(add brain activity by pain, blood loss and fractures)

Добавлена депрессия, фикс нескольких багов, нормально отбалансированное повышение мозговой активности от боли, регуляция ее потерей крови.
Переделана шизофрения, теперь она в одних местах сильнее, в других слабее. Депрессию можно получить с шансом 5% при передозе галоперидолом(нарколептическая депрессия)

TapKo4eB commented on code/__defines/dna.dm:
```diff
@@ -29,11 +29,13 @@
-#define COUGHING    0x4
-#define TOURETTES   0x8
-#define NERVOUS     0x10
+#define NEARSIGHTED 1
+#define EPILEPSY    2
+#define COUGHING    4
+#define TOURETTES   8
+#define NERVOUS     16
+#define SCHIZOPHRENIA 32
+#define DEPRESSION    64
```
а чем тебе не понравились старые бит флаги

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+			return "Significant"
+	return "Moderate"
+
+/proc/get_brain_status(var/brain_damage_procents)
```
удоляй пока не зарезал
```suggestion
/proc/get_brain_status(brain_damage_procents)
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+		return "Significant"
+	return "Moderate"
+
+/proc/get_blood_status(var/procents)
```
зарежу удаляй пока не
```suggestion
/proc/get_blood_status(procents)
```

TapKo4eB commented on code/modules/surgery/surgery.dm:
```diff
@@ -32,11 +32,6 @@
 	if(!hasorgans(target))
 		return 0
 
-	if(allowed_species)
-		for(var/species in allowed_species)
-			if(target.species.name == species)
-				return 1
-
```
если удаляешь использование то и дефайн листа удали из датума операций

kcalbCube replied:
Не понял

TapKo4eB reviewed (commented):
ревью кода

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
 	data["scan_date"] = stationtime2text()
-	data["brain_activity"] = null
-	data["pulse"] = null
+
```
```suggestion

```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
 		var/obj/item/organ/internal/brain/brain = H.internal_organs_by_name[BP_BRAIN]
 		if (!brain || H.stat == DEAD || (H.status_flags & FAKEDEATH))
-			data["brain_activity"] = 0
+			data["brain_activity"]    = 0
```
```suggestion
			data["brain_activity"] = 0
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
-				data["brain_activity"] = 1 - (brain.damage / brain.max_damage)
-	else
-		data["brain_activity"] = null
+			data["brain_activity"]    = brain.get_activity()
```
```suggestion
			data["brain_activity"] = brain.get_activity()
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+		data["brain_activity_status"] = src.get_brain_activity_status()
+
+		var/brain_damage_procents = 100*(brain.damage / brain.max_damage)
+		data["brain_damage"]     	  = 100-brain_damage_procents
```
```suggestion
		data["brain_damage"] = 100-brain_damage_procents
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+
+		var/brain_damage_procents = 100*(brain.damage / brain.max_damage)
+		data["brain_damage"]     	  = 100-brain_damage_procents
+		data["brain_status"]          = get_brain_status(100-brain_damage_procents)
```
```suggestion
		data["brain_status"] = get_brain_status(100-brain_damage_procents)
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+		var/brain_damage_procents = 100*(brain.damage / brain.max_damage)
+		data["brain_damage"]     	  = 100-brain_damage_procents
+		data["brain_status"]          = get_brain_status(100-brain_damage_procents)
+
```
```suggestion

```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
-			data["pulse"] = H.get_pulse(1)
-	else
-		data["pulse"] = null
+			var/pulse     = text2num(H.get_pulse(1))
```
```suggestion
			var/pulse = text2num(H.get_pulse(1))
```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -597,4 +590,20 @@
-	. = jointext(dat,"<br>")
\ No newline at end of file
+	. = jointext(dat,"<br>")
+
```
```suggestion

```

TapKo4eB commented on code/game/machinery/flasher.dm:
```diff
@@ -104,6 +106,7 @@
 			O.Stun(flash_time / 2)
 			O.Weaken(3)
 
+
```
```suggestion

```

Commit:
Update code/game/machinery/adv_med.dm

fix stupid error

Co-authored-by: TapKo4eB <48761132+TapKo4eB@users.noreply.github.com>

TapKo4eB commented on code/modules/item_worth/reagents.dm:
```diff
@@ -600,4 +600,13 @@
+	value = 3.5
+
+/datum/reagent/calciumxon
+	value = 3.5
```
```suggestion
	value = 3.5

```

TapKo4eB commented on code/modules/maps/swapmaps.dm:
```diff
@@ -624,46 +624,51 @@ proc/SwapMaps_LoadChunk(chunk_id,turf/locorner)
 	if(text)
-		S=new
-		S.ImportText("/",file("map_[id].txt"))
+		S  =new
```
```suggestion
		S = new
```

TapKo4eB commented on code/modules/organs/internal/brain.dm:
```diff
@@ -48,11 +61,10 @@
-	max_damage = 100
-	if(species)
-		max_damage = species.total_health/2
+	max_damage         = species? (species.total_health/2) : 100
```
```suggestion
	max_damage = species? (species.total_health/2) : 100
```

TapKo4eB commented on code/modules/organs/internal/brain.dm:
```diff
@@ -48,11 +61,10 @@
+
 	min_bruised_damage = max_damage*0.25
-	min_broken_damage = max_damage*0.75
+	min_broken_damage  = max_damage*0.75
```
```suggestion
	min_broken_damage = max_damage*0.75
```

TobyThorne replied:
шо за пробел

TapKo4eB commented on code/modules/organs/internal/brain.dm:
```diff
@@ -175,23 +189,29 @@
 						owner.visible_message("<B>[owner]</B> faints!", \
-											  SPAN("warning", "You feel extremely [pick("dizzy","woozy","faint")]..."))
-						owner.Paralyse(3,5)
+											  SPAN_WARNING("You feel extremely [pick("dizzy","woozy","faint")]..."))
```
```suggestion
					SPAN_WARNING("You feel extremely [pick("dizzy","woozy","faint")]..."))
```

TapKo4eB commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -114,16 +114,4 @@
-
-/datum/reagent/proc/reaction_mob(mob/target)
-	touch_mob(target)
+	holder = null
```
```suggestion
	holder = null

```

TapKo4eB commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1043,3 +1043,85 @@
+			continue
+		if(E.limb_flags & ORGAN_FLAG_CAN_BREAK)
+			E.heal_damage(0.5)
+			E.update_damstate()
```
```suggestion
			E.update_damstate()

```

TapKo4eB commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -2098,3 +2104,15 @@
+	name = "Calciumxon"
+	result = /datum/reagent/calciumxon
+	required_reagents = list(/datum/reagent/drink/milk = 5, /datum/reagent/dermaline = 1)
+	result_amount = 2
```
```suggestion
	result_amount = 2

```

TapKo4eB commented on nano/templates/adv_med.tmpl:
```diff
@@ -0,0 +1,227 @@
+            </tr>
+            {{/for}}
+    </table>
+</div>
```
```suggestion
</div>

```

TapKo4eB commented on code/modules/organs/internal/brain.dm:
```diff
@@ -200,38 +220,95 @@
+				to_chat(owner, SPAN_WARNING("[pick("You feel a sad", "You lost", "Where has happiness gone?")]"))
+			if(11)
+				to_chat(owner, SPAN_DEADSAY("[pick("You want to die", "You want to end a meaningless existence")]"))
+
```
```suggestion

```

TapKo4eB commented on code/game/machinery/adv_med.dm:
```diff
@@ -281,109 +259,127 @@
+				data["pulse_status"] = "Moderate"
 
-	data["blood_type"] = null
+
+
```
```suggestion
```

TapKo4eB reviewed (commented):
ревью кодстайла

kcalbCube commented on nano/templates/adv_med.tmpl:
```diff
@@ -0,0 +1,227 @@
+        </div>
+     {{/if}}
+
+     {{for data.systemStatus}}
+        <div>
+            <div class="itemLabel" style="overflow: hidden; max-width: 5x;">
+                {{:value[0]}}:
+            </div>
+            <div class="{{:value[1]}}" style="float: left">
+                {{:value[1]}}
+            </div><br>
+        </div>
+     {{/for}}
+ </div>
```
???

TapKo4eB replied:
каждый <div> должен иметь класс итем
а на деле только первый

kcalbCube replied:
Да вроде все нормально.

TapKo4eB replied:
ладно похрен и так работает

Commit:
fix(codestyle)

Commit:
Merge branch 'dev' of https://github.com/kcalbCube/OnyxBay_medupdate into dev

Commit:
balance change(disability give)

Теперь для получения дизабилити нужно не 95% повреждений, а 75%

kcalbCube commented:
@TobyThorne ready to review

TobyThorne commented:
а можно пожалуйста ещё больше разных фич в ПР

TobyThorne commented on code/modules/item_worth/reagents.dm:
```diff
@@ -564,7 +564,7 @@
 
 /datum/reagent/soporific
-	value = 2.5
+	value = 3.5
```
Почему слиптоксин стал дороже хлорки, зачем потрогано вино, почему сахар стоит две тыщи рублей за килограмм, нихуя не понимаю.

kcalbCube replied:
Ладно, хуйню здесь сделал.

kcalbCube replied:
Как убрать 🤔🤔🤔

kcalbCube replied:
Удалил

kcalbCube commented:
А надо?

Commit:
remove(worth change) 

Удалён маразм.

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -1,237 +1,314 @@
+	if(damage > 0.3*max_damage && prob(1))
+		owner.custom_pain("Your feel strong pain.", 25)
+	if(damage > 0.5*max_damage && prob(1))
+		owner.custom_pain("Your feel a lot of pain.", 40)
```
т.е. за один тик космонавтик с 50+ дамага в мозге может с определённым шансом получить в чат
Your feel pain.
Your feel strong pain.
Your feel a lot of pain.
Одновременно. "Your feel". Без индикации, что болит именно голова.

kcalbCube replied:
Болит не голова, болит случайная часть тела

TobyThorne commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -734,7 +732,8 @@
 	..()
-	M.add_chemical_effect(CE_PULSE, 2)
+	M.add_chemical_effect(CE_PULSE, 3)
+	M.add_chemical_effect(CE_BREATHLOSS, 2)
```
Зачем обычному передозу никотином удушение на уровне героина с водкой?

kcalbCube replied:
Ну, в жизни если ты покурил восемь пачек то удушье точно появится

TobyThorne commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1043,3 +1043,85 @@
+
+/datum/reagent/calciumxon
+	name = "Calciumxon"
+	description = "Psyloxon is a antiepileptical drug. Stabilizes brain activity."
```
name = "Calciumxon"
description = "Psyloxon is a antiepileptical drug.

TobyThorne commented on code/modules/surgery/surgery.dm:
```diff
@@ -6,7 +6,6 @@
 	var/list/allowed_tools = null
 	// type paths referencing races that this step applies to.
-	var/list/allowed_species = null
 	var/list/disallowed_species = null
```
allowed_species удалил
@ 
disallowed_species оставил на память

kcalbCube replied:
Какой то баг. Наверное ctrl z лишний раз нажал

TobyThorne commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1043,3 +1043,85 @@
+			continue
+		if(E.limb_flags & ORGAN_FLAG_CAN_BREAK)
+			E.heal_damage(0.5)
+			E.update_damstate()
```
Как это вообще в теории должно убрать перелом?

kcalbCube replied:
Перелом реализован как сломанный орган с пометкой о том что это перелом. Орган лечится и пометка обновляется - перелом уходит.

TobyThorne commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1043,3 +1043,85 @@
+
+/datum/reagent/galoperidol/overdose(mob/living/carbon/M, alien)
+	M.add_chemical_effect(CE_SLOWDOWN, 20)
+	M.add_chemical_effect(CE_BREATHLOSS, 10)
```
Замедление до уровня статичного объекта и мгновенные десятки оксидамага с передоза. Нашли новую хлорку на максималках.

kcalbCube replied:
Окей, повышу шанс на развитие депрессии и уберу это.

Commit:
fix(galoperidol overdose)

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -1,237 +1,314 @@
+			if(1 to 3)
+				owner.emote("cry")
+			if(4 to 10)
+				to_chat(owner, SPAN_WARNING("[pick("You feel a sad", "You lost", "Where has happiness gone?")]"))
```
a sad


bashar

Commit:
fix(Calciumxon description)

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -1,237 +1,314 @@
+		owner.eye_blurry = 10
+	if(get_activity() <= 40 && prob(1))
+		owner.slurring = max(owner.slurring, 2)
+	if(get_activity() <= 30 && prob(1) && owner.get_active_hand())
```
Десять миллиардов вызовов get_activity у каждого человечка каждый тик. Выглядит надёжно.

TobyThorne reviewed (changes requested):
Короче, ставлю на то, что это вообще не тестилось.

TobyThorne commented:
> #3691
> Я предлагаю добавить автору пула решение на это предложение

С такими идеями больше не приходи, пожалуйста.

kcalbCube commented:
> Короче, ставлю на то, что это вообще не тестилось.

Ну, я долго тестировал (7 часов наберётся) такая себе в view variables различные варианты. Это не считается?

Commit:
rework(psychiatry) remove(tgui to nanoui migration) fix(operation scanner) add(psychiatry module) add(depression, schizophrenia, ocd) fix(some brain activity fixes) remove(turrets and nervous disabilities) add(glob.epilepsyblock) fix(galoperidol overdose)

Все в описании. Вынес мусорный перенос на наноуи в окно, создал модуль психиатрии, убрал другой ненужный мусор и прочее.

kcalbCube commented:
Проиграл не тот кто упал, проиграл тот кто не поднялся. @TobyThorne ready for review

Destrolaric commented:
Here we go again

kcalbCube commented:
@TobyThorne жду разьеба по коду, уже джва дня.

TobyThorne commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -32,6 +32,10 @@
 
+		if(ishuman(src))
+			var/mob/living/carbon/human/H = src
+			H.handle_psychiatry()
```
зачем это в life карбонов с проверкой, когда можно просто засунуть в life хуманов

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -1,3 +1,4 @@
+#define DISABILITY_GIVE_DELAY 2*60*10
```
магические цыферки

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -20,6 +21,16 @@
+	var/last_schizophrenia_attack = 0
+	var/schizophrenia_delay       = 0
+
+// in procents
```
что такое procents

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -37,9 +48,9 @@
-	icon = 'icons/mob/alien.dmi'
+	name       = "thinkpan"
+	desc       = "It looks kind of like an enormous wad of purple bubblegum."
+	icon 	   = 'icons/mob/alien.dmi'
```
что это вообще такое

4310V343k replied:
Везде так. Я уже вроде проходился по этому пиздецу

Commit:
fix(reagents worth) 

Что-то сломалось в коммитах, вернул как надо.

TobyThorne commented on code/modules/organs/internal/brain.dm:
```diff
@@ -175,23 +188,29 @@
 					if(!owner.weakened && prob(10))
 						owner.Weaken(rand(1,3))
-						to_chat(owner, SPAN("warning", "You feel [pick("dizzy","woozy","faint")]..."))
+						to_chat(owner, SPAN_WARNING("You feel [pick("dizzy","woozy","faint")]..."))
```
сейчас бы менять нормальные макросы на эту укрохуйню

kcalbCube replied:
??? Чем плох SPAN_WARNING?

TobyThorne commented:
В хер всё это, короче. Проверю код целиком, если остальные геймдизы аппрувнут. Я голосую против (по крайней мере, в том состоянии, в котором оно находится в данный момент), ибо:
- Усложнения ради усложнений. Медбей в каждом первом раунде и так напоминает рынок в чёрную пятницу.
- Из расстройств хоть какой-то интерес для геймплея представляет разве что шизофрения. Если бы добавил ещё что-то смешное вроде туретта, нарушений речи, каких-нибудь тиков с рандомным выбрасыванием удерживаемого предмета, телепатию/разговоры с призраками (с мизерным шансом, ессесна) и всякого прочего, что хоть как-то влияет на раунд - я бы, наверое, аппрувал не глядя. Но выборка из галюнов, рандомных убивающих станов и спама в чатике - какая-то хуйня из-под коня.
- Расстройства должны даваться хотя бы с каким-то шансом, а не гарантированно. Желательно - чтоб меньше 1:1. А то любое ЧП - и станция полна шизиков.
- Должен быть способ перманентно лечить расстройства. Какие-нибудь мозговые импланты из РнД, артефакты с шахты, вытяжки из малиновых в крапинку слаймов (хоть для чего-то были бы нужны), отвары из форона, мухоморов и мочи розовой таяры. 26 век, ёптыть.
- Огромная куча разношёрстных изменений в ПРе. Тот же душащий покруче петли передоз никотином - это хуерга какая-то. Или реагент, который должен лечить кости. Те же кости надо отдельно реворкать, а не приделывать к ним нерабочий костыль (иронично, однако).

kcalbCube commented:
Вас понял.

Destrolaric commented:
![AAAc2iwZy6ib7ssty5IB-SjFqcBXmcr7i5jgIlx4cWCILiEQivqYiUFlRrqwreV4hhueHtkGrU6ghf0XLDK51YRp-24](https://user-images.githubusercontent.com/20579256/100103386-e13f3780-2e75-11eb-83d0-79550f1d54d5.jpg)
ПР не болей

kcalbCube commented:
> ![AAAc2iwZy6ib7ssty5IB-SjFqcBXmcr7i5jgIlx4cWCILiEQivqYiUFlRrqwreV4hhueHtkGrU6ghf0XLDK51YRp-24](https://user-images.githubusercontent.com/20579256/100103386-e13f3780-2e75-11eb-83d0-79550f1d54d5.jpg)
> ПР не болей
> 

Приложу через 7 часов, вдруг поможет.

YouBro commented:
Слишком много лишних нововведений. Не трогай медицину.

kcalbCube commented:
Исходный код был случайно удален. Нужна реорганизация.

Destrolaric commented:
lol

kcalbCube commented:
жалко девочку.

kcalbCube commented:
@Doster-d 
> > ты чё еблан, нахуя ты мыслишь уже той самой знакомой с лысым из бразерс
> 
> Чё? Понял хоть сам что сказал?
> 
> > все эти правила они абстрактные, они создают представление про идеальный код.
> 
> Ну да, и мой код не нарушает эти правила.
> 
> > любая лишняя вложенность это +1 символ на строку, и -4 символа к пространству для просмотра, редактирования и написания кода.
> 
> Ты читать умеешь? Где ебать эта хуйня мешает тебе редактировать/просматривать говнокод, если там ебать ВСЕГО ОДНА ВЛОЖЕННОСТЬ НА ИФЕ?!
> 
> > а по поводу комментариев... ты мне говорил что комментарий урод, что он из детдома с лошадимым рылом, и вообще оникс девелопер, а теперь включаешь заднюю. так не делается
> 
> Ты еблан, мне было бы абсолютно похуй на комментарии, хоть ты бы писал их арабскими символами, если бы это не было бы в провер очках.
> 
> Если тебе это не нравится, го энд билд юр гейм оф зе дрим виз гитхаб рулез оф зе дрим.

ты хочешь сказать что ты заменял собой работу какой то там машины, и не представлял более никакой ценности?

kcalbCubeFork commented:
так у нас есть флешка с потерянными сорсами, покс которого не существует, две тысячи терминов, экспонента от минус пи, убивающий за полчаса порез от бумаги, тридцать видов наркотиков, тромб пятки, спидран по остановке жопы в сердце и
этого больше нет на ониксе и я прошу выдать семибоярщине временный разбан что бы мы смогли увидится со старыми друзьяшками

Doster-d commented:
Вы ошиблись репой, пожалуйста, не пишите сюда больше, от вас спиртом воняет :)

DarkHunter7756 commented:
Тридцать видов наркотиков даже я не против увидеть 😼

kcalbCubeFork commented:
> Тридцать видов наркотиков даже я не против увидеть smirk_cat

ты мое приглашение отклонил

kcalbCubeFork commented:
я бросил, в том числе кринж парады, хочу исполнить долг перед обществом. подскажите что вам нужно накодить.

Doster-d commented:
> я бросил, в том числе кринж парады, хочу исполнить долг перед обществом. подскажите что вам нужно накодить.

"Я наплевал вам в колодец, Вы закрыли от меня колодец, и я прошу дать мне снова подойти к колодцу"

kcalbCubeFork commented:
> > я бросил, в том числе кринж парады, хочу исполнить долг перед обществом. подскажите что вам нужно накодить.
> 
> "Я наплевал вам в колодец, Вы закрыли от меня колодец, и я прошу дать мне снова подойти к колодцу"

на твоем месте я бы так не говорил, я ведь знаю что ты есть на кубоконфе, я сам тебя пригласил.
кстати не закрыли мне колодец, мне нужно чем-то закрыть появившиеся пробелы в жизни. ну и я нормально вроде говорю

Doster-d commented:
Ой, ой, ой, я потеряю доступ к конфе человека, который плюет в колодцы, как это жалко! Неужели мне надо вести диалог с автором ВЕЛИКОЙ кубомедициной как с МУЗОЙ?!
А что ты тогда с мультиакка сидишь, дорогой, если тебе не закрыли доступ к нашему колодцу?
Говоришь ты нормально, только ты не понимаешь, что ты тут юродивый для всех из-за твоих действий в прошлом, и это тебе, вероятно, никогда не забудут.

Хочешь попытаться изменить такое отношения? Фикси баги. Причем много тебе надо будет пофиксить, чтобы ты хотя бы стал рукопожатым.

kcalbCubeFork commented:
Достер, всё будет хорошо.

No-Tips commented:
Впили реплеи и мы в расчете

Zert0X commented:
> Впили реплеи и мы в расчете

так курва, не трогать мой хлеб

Nod404 commented:
> Говоришь ты нормально, только ты не понимаешь, что ты тут юродивый для всех из-за твоих действий в прошлом

я люблю куба

NefrNerf commented:
Если хочешь помогать, то окей, из большого:
Культ, портировать часовщиков или лучше допилить наш до адекватного (портировать и почистить)
Вампиры, допилить до нормального состояние(сделать их неживыми и так далее)
Можешь какого антага на мидгейм портануть, а так баги всегда есть.

kcalbCubeFork commented:
> Если хочешь помогать, то окей, из большого: Культ, портировать часовщиков или лучше допилить наш до адекватного (портировать и почистить) Вампиры, допилить до нормального состояние(сделать их неживыми и так далее) Можешь какого антага на мидгейм портануть, а так баги всегда есть.

подожду разработчиков.

Zert0X commented:
> > Если хочешь помогать, то окей, из большого: Культ, портировать часовщиков или лучше допилить наш до адекватного (портировать и почистить) Вампиры, допилить до нормального состояние(сделать их неживыми и так далее) Можешь какого антага на мидгейм портануть, а так баги всегда есть.
> 
> подожду разработчиков.

А чё нас ждать? Вкладочка иссуи с гитхаба никуда не пропала.

Zert0X commented:
> Вампиры, допилить до нормального состояние(сделать их неживыми и так далее)
Where is issue?!

kcalbCubeFork commented:
> > > Если хочешь помогать, то окей, из большого: Культ, портировать часовщиков или лучше допилить наш до адекватного (портировать и почистить) Вампиры, допилить до нормального состояние(сделать их неживыми и так далее) Можешь какого антага на мидгейм портануть, а так баги всегда есть.
> > 
> > 
> > подожду разработчиков.
> 
> А чё нас ждать? Вкладочка иссуи с гитхаба никуда не пропала.

будто ты не знаешь что я имею в виду.

TobyThorne commented:
хули вы тут чатик устроили
